lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -488,8 +489,10 @@ class TraceProcessor {
     // Find the inspected frame
     const mainFrameIds = this.findMainFrameIds(keyEvents);
 
-    // Filter to just events matching the frame ID for sanity
-    const frameEvents = keyEvents.filter(e => e.args.frame === mainFrameIds.frameId);
+    // Filter to just events matching the main (top-level) frame ID for sanity, unless we're
+    // piercing iframes
+    const frameEvents = settings.pierceIframes ?
```

Comment:
I'm a little nervous about all the other assumptions that might fall through with this change, the flag feels a little less like `pierceIframes` and a little more like `--look-at-all-events`, which if the user is always running just a single tab in their chrome instance is probably the same thing, but definitely dangerous in other situations.

Maybe this just means we very carefully document usage of this flag?